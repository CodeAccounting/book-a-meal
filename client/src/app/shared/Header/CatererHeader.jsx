import React, { Fragment } from 'react';
import moment from 'moment';

/**
 * @exports
 * @function CatererHeader
 * @param {object} props
 * @returns {JSX} CatererHeader
 */
const CatererHeader = () => (
  <Fragment>
    <div className="menu-control d-none-md">
      <a href="#sidenav" className="d-none-md">
        <img src="./img/toggler.png" alt="toggler" />
      </a>
    </div>
    <div className="page-title d-flex-md">
      <div className="date">
        <h2>{moment().format('dddd[,] Do MMMM YYYY')}&nbsp; &nbsp; {moment().format('HH:mm')}</h2>
      </div>
    </div>
  </Fragment>
);

export default CatererHeader;
