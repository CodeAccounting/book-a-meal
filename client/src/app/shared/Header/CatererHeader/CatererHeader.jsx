import React, { Fragment } from 'react';
import moment from 'moment';
import PropTypes from 'prop-types';

/**
 * @exports
 * @function CatererHeader
 * @param {object} props
 * @returns {JSX} CatererHeader
 */
const CatererHeader = ({ currentDay, showTime, dateType }) => {
  const date = dateType === 'menu' ? moment(currentDay).format('dddd[,] Do MMMM YYYY') : moment().format('dddd[,] Do MMMM YYYY');

  return (
    <Fragment>
      <div className="menu-control d-none-md">
        <a href="#sidenav" className="d-none-md">
          <img src="./img/toggler.png" alt="toggler" />
        </a>
      </div>
      <div className="page-title d-flex-md">
        <div className="date">
          <h2>{date}&nbsp; &nbsp; {showTime && moment().format('HH:mm')}</h2>
        </div>
      </div>
    </Fragment>
  );
};

CatererHeader.propTypes = {
  currentDay: PropTypes.string.isRequired,
  showTime: PropTypes.bool,
  dateType: PropTypes.string
};

CatererHeader.defaultProps = {
  showTime: true,
  dateType: null
};

export default CatererHeader;
