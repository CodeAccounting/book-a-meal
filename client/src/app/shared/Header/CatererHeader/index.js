import { connect, } from 'react-redux';
import CatererHeader from './CatererHeader';

const mapStateToProps = state => ({
  currentDay: state.menu.currentDay
});

export default connect(mapStateToProps)(CatererHeader);
