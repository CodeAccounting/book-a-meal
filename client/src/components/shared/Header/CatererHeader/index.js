import { connect, } from 'react-redux';
import { bindActionCreators } from 'redux';
import CatererHeader from './CatererHeader';
import { toggleSideNav } from '../../../../store/actions/ui';

const mapStateToProps = state => ({
  currentDay: state.menu.currentDay
});

const mapDispatchToProps = dispatch => bindActionCreators({
  toggleSideNav
}, dispatch);

export default connect(mapStateToProps, mapDispatchToProps)(CatererHeader);
