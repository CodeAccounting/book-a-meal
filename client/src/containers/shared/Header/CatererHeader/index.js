import { connect, } from 'react-redux';
import { bindActionCreators } from 'redux';
import CatererHeader from '../../../../components/shared/Header/CatererHeader';
import { toggleSideNav } from '../../../../actions/ui';

const mapStateToProps = state => ({
  currentDay: state.menu.currentDay
});

const mapDispatchToProps = dispatch => bindActionCreators({
  toggleSideNav
}, dispatch);

export default connect(mapStateToProps, mapDispatchToProps)(CatererHeader);
