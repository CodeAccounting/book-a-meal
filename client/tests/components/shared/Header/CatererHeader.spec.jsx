import React from 'react';
import CatererHeader from '../../../../src/app/shared/Header/CatererHeader';


describe('CatererHeader', () => {
  it('renders correctly', () => {
    const { now } = Date;
    Date.now = jest.fn(() => 0); // mock Date now

    try {
      const shallowWrapper = shallow(<CatererHeader />);

      expect(toJson(shallowWrapper)).toMatchSnapshot();
    } finally {
      Date.now = now; // restore original now method on Date object
    }
  });
});
