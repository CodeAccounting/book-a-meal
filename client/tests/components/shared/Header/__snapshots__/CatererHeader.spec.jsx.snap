// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CatererHeader renders correctly 1`] = `
<React.Fragment>
  <div
    className="menu-control d-none-md"
  >
    <a
      className="d-none-md"
      href="#sidenav"
    >
      <img
        alt="toggler"
        src="./img/toggler.png"
      />
    </a>
  </div>
  <div
    className="page-title d-flex-md"
  >
    <div
      className="date"
    >
      <h2>
        Thursday, 1st January 1970
            
        00:00
      </h2>
    </div>
  </div>
</React.Fragment>
`;
